convert2rhel/checks.py

```diff
@@ -57,6 +62,88 @@ def perform_pre_ponr_checks():
     ensure_compatibility_of_kmods()
 
 
+def check_convert2rhel_latest():
+    """Make sure that we are running the latest version of convert2rhel"""
+
+    # get lastest avalible version
+
+    repo_content = (
+        "[convert2rhel]\n"
+        "name=Convert2RHEL Repository\n"
+        "baseurl=https://ftp.redhat.com/redhat/convert2rhel/$releasever/os/\n"
+        "gpgcheck=0\n"
+        "enabled=1\n"
+    )
+
+    repo_dir = os.path.join(utils.TMP_DIR, "convert2rhel_repo")
+    utils.mkdir_p(repo_dir)
+    # repo_dir = os.path.join("/tmp/","convert2rhel_repo")
+    repo_path = os.path.join(repo_dir, "convert2rhel.repo")
+    store_content_to_file(filename=repo_path, content=repo_content)
+
+    # repoquery --setopt=reposdir=$(pwd)/test_repo --releasever=8 convert2rhel
+    raw_output_convert2rhel_versions, _ = run_subprocess(
+        "repoquery --releasever={releasever} --setopt=reposdir={repodir} convert2rhel".format(
+            releasever=system_info.releasever,
+            # releasever="8",
+            repodir=repo_dir,
+        ),
+        print_output=False,
+    )
+
+    #   - if cannot access url: print warning and continue
+    PKG_NEVR = r"\b(\S+)-(?:([0-9]+):)?(\S+)-(\S+)\b"
+    convert2rhel_versions = re.findall(PKG_NEVR, raw_output_convert2rhel_versions, re.MULTILINE)
+
+    latest_version = ("0", "0.00", "0")
+    #   - get latest version from https://ftp.redhat.com/redhat/convert2rhel/.
+    for package_version in convert2rhel_versions:
+        if package_version[0] != "convert2rhel":
+            continue
+
+        ver_compare = rpm.labelCompare(package_version[1:], latest_version)
+
+    # if current version is not latest:
+    ver_compare = rpm.labelCompare(("0", convert2rhel_version, "0"), ("0", latest_version[1], "0"))
```

Comment:
This could still be changed to reduce confusion

Reply:
I have changed the format of the latest_version var in this line!

Reply:
I have just ran the conversion with these's changes and have realized that how the version compare is set up now it is expecting latest_version to have three segments like this ("0", "0.00", "0") which is why I had it set up like that. Should I write a comment as to why it needs to be in that order or should I try and see if I can change it to work with the format of just the version (0.00)

Reply:
@Andrew-ang9 At this stage, since you are accessing `latest_version[1]` we still have the same format as before when you first initialized the variable `latest_version = ("0", "0.00", "0")`. Unless the `latest_version` variable format is changed when you assign it the `package_version[1:]` it should work fine doing it like this and it will cause less confusion

What is the variable format that you are seeing? What is it that isn't working?